staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -120,7 +120,26 @@ func newOIDCAuthProvider(_ string, cfg map[string]string, persister restclient.A
 
 	// Check cache for existing provider.
 	if provider, ok := cache.getClient(issuer, clientID); ok {
-		return provider, nil
+		provider.mu.Lock()
+		defer provider.mu.Unlock()
```

Comment:
in case useCache is false, this defer unlock will cause issue in SetClient call where it tries locking again.

Reply:
I separate it into a function. PTAL, thanks